data/slash/multipleHeaderSigningProof.go

```diff
@@ -36,12 +46,24 @@ func (m *MultipleHeaderSigningProof) GetHeaders(pubKey []byte) []data.HeaderHand
 		return nil
 	}
 
-	headersV2, exist := m.HeadersV2[string(pubKey)]
-	if !exist {
+	slashData, exists := m.SignersSlashData[string(pubKey)]
+	if !exists {
 		return nil
 	}
 
-	return headersV2.GetHeaderHandlers()
+	idx := uint32(0)
+	bitmap := slashData.GetSignedHeadersBitMap()
+	headers := m.HeadersV2.GetHeaderHandlers()
+
+	ret := make([]data.HeaderHandler, 0)
+	for _, header := range headers {
```

Comment:
for idx, header := range headers ?
then remove idx init and idx++

Reply:
Applied your suggestion.